### PR TITLE
Issue 817: Difficult to scroll when drawing free-hand

### PR DESCRIPTION
**Pull request by tdakho**
WJE/Plannotate2/issues/817

Also WJE/Plannotate2/pull/1109

### CODE REVIEW

File: Smooth Line View/SmoothLineView.m [hunk 1]
```diff
@@ -57,49 +60,93 @@ + (Class)layerClass
...
+    
+    _bgColor = backgroundColor;
+}
+
+- (void) drawRect:(CGRect)rect {
+    // clear rect
+    // we use an internal property because drawRect is called from a background thread
+    // and throws a warning if you try to access [UIView backgroundColor] while
+    // not on the main thread.
+    [self.bgColor set];
```
Comment (wrkhenddher) [inline Smooth Line View/SmoothLineView.m:138]: never seen this construct ... where is this documented?

File: Smooth Line View/SmoothLineView.m [hunk 2]
```diff
@@ -57,49 +60,93 @@ + (Class)layerClass
...
+- (id) initWithFrame:(CGRect)frame andExistingView:(SmoothLineView*)view;
+{
+    self = [self initWithFrame:frame];
+    
+    if (self)
+    {
+        _pathSnapshots = [view.pathSnapshots mutableCopy];
+        _path = CGPathCreateMutableCopy(view.path.CGPath);
+        self.bzPath = [UIBezierPath bezierPathWithCGPath:_path];
+        self.renderAsArea = view.renderAsArea;
```
Comment (wrkhenddher) [inline Smooth Line View/SmoothLineView.m]: remember `view.bgColor`?

**Review by wrkhenddher — COMMENTED** [review]
Did you try `Done as many` and `Done`

**Comment by tdakho** [comment]
I did

File: Smooth Line View/SmoothLineView.m [hunk 3]
```diff
@@ -265,6 +265,8 @@ - (UIBezierPath*) path
 
 - (void) setPath:(UIBezierPath*) bezierPath
 {
+    CGMutablePathRef oldPath = _path;
+    CGPathRelease(oldPath);
```
Comment (wrkhenddher) [inline Smooth Line View/SmoothLineView.m]: Perhaps `NSAssert(bezierPath, ...)` to make sure ... I don't know what would happen with `CGPathCreateMutableCopy(bezierPath.CGPath)`

**Pull request merged**